### PR TITLE
Set up associations review

### PR DESCRIPTION
**Pull request by akezeth**
- Created a class Classroom.
- Created the has-many/belongs-to relationship between Classroom and Student.
- Created a class Book.
- Created a class Rental.
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline book.rb]: - [ ] there should be a method `add_rental(person, date)` which takes person and date as arguments, kindly add that

File: person.rb [hunk 2]
```diff
@@ -2,14 +2,15 @@
 
 class Person < Nameable
   attr_reader :id
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
```
Comment (VICTOR-CHUKWU) [inline person.rb:5]: - [ ] there should be a method `add_rental(book, date)` which takes book and date as arguments, kindly add that

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Rental
+  attr_accessor :date
```
Comment (VICTOR-CHUKWU) [inline rental.rb]: - [ ] book and person should also be made accessible :+1:

File: student.rb [hunk 4]
```diff
@@ -1,9 +1,12 @@
 require_relative 'person'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(classroom, age, name = 'unknown', parent_permmission: true)
     super(age, name, parent_permmission)
     @classroom = classroom
+    classroom.students.push(self) unless classroom.students.include?(self)
```
Comment (VICTOR-CHUKWU) [inline student.rb]: - [ ] According to project requirement this should be inside a method called classroom and not inside your initialize, kindly fix this, your method should look like this
```ruby
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @akezeth  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @akezeth , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Set up associations ✔️
- Well Structured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**